daemon/execdriver/native/create.go

```diff
@@ -222,3 +227,17 @@ func (d *driver) setupLabels(container *libcontainer.Config, c *execdriver.Comma
 
 	return nil
 }
+
+func (d *driver) setupAdditionalGroups(container *libcontainer.Config, c *execdriver.Command) error {
+	container.AdditionalGroups = make([]int, len(c.GroupAdd))
+
+	for i, g := range c.GroupAdd {
+		group, err := strconv.Atoi(g)
```

Comment:
But the host's groups file is pretty meaningless inside a container,
especially one we have user ns.

I guess I am arguing simpler is better.

On Wed, Feb 11, 2015 at 4:01 PM, Mrunal Patel <notifications@github.com>
wrote:

> In daemon/execdriver/native/create.go
> <https://github.com/docker/docker/pull/10717#discussion_r24546903>:
>
> > @@ -222,3 +227,17 @@ func (d *driver) setupLabels(container *libcontainer.Config, c *execdriver.Comma
> >
> >  	return nil
> >  }
> > +
> > +func (d *driver) setupAdditionalGroups(container *libcontainer.Config, c *execdriver.Command) error {
> > +	container.AdditionalGroups = make([]int, len(c.GroupAdd))
> > +
> > +	for i, g := range c.GroupAdd {
> > +		group, err := strconv.Atoi(g)
>
> I think they would have to be from the host's groups file. They could be
> picked from the container but different containers could have same group
> names mapped to different gids and that kind of defeats the purpose here.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/10717/files#r24546903>.
>